tests/test_graph/test_algorithms.py

```diff
@@ -0,0 +1,180 @@
+
+
+@pytest.mark.parametrize(
+    ("mean", "stdev", "samples", "rtol"),
+    [
+        pytest.param(1.0, 1.0, 10, 1, id="N(mean=1, stdev=1), 10 samples"),
+        pytest.param(2.0, 0.8, 1000, 1e-1, id="N(mean=2, stdev=0.8), 1000 samples"),
+        pytest.param(1.0, 0.8, 100000, 1e-2, id="N(mean=1, stdev=0.8), 10^5 samples"),
+        pytest.param(1.0, 1.2, 10000000, 1e-3, id="N(mean=1, stdev=1.2), 10^7 samples"),
+    ],
+)
+def test_mean_stdev_single_normal_node(samples, rtol, mean, stdev, rng_key):
+    node = DistributionNode(
+        Normal,
+        label="X",
+        constant_parameters={"loc": mean, "scale": stdev},
+    )
+
+    graph = Graph(label="G0")
+    graph.add_node(node)
+
+    # To compensate for rng-key splitting in sample methods, note the "split" key
+    # that is actually used to draw the samples from the distribution, so we can
+    # attempt to replicate its behaviour explicitly.
+    key = jax.random.split(rng_key, 1)[0]
+    what_we_should_get = jax.random.multivariate_normal(
+        key, jax.numpy.atleast_1d(mean), jax.numpy.atleast_2d(stdev**2), shape=samples
+    )
+    expected_mean = what_we_should_get.mean()
+    expected_std_dev = what_we_should_get.std()
+
+    # Check within hand-computation
+    assert np.isclose(
+        causalprog.algorithms.expectation(
+            graph, outcome_node_label="X", samples=samples, rng_key=rng_key
+        ),
+        mean,
+        rtol=rtol,
+    )
+    assert np.isclose(
+        causalprog.algorithms.standard_deviation(
+            graph, outcome_node_label="X", samples=samples, rng_key=rng_key
+        ),
+        stdev,
+        rtol=rtol,
+    )
+    # Check within computational distance
+    assert np.isclose(
+        causalprog.algorithms.expectation(
+            graph, outcome_node_label="X", samples=samples, rng_key=rng_key
+        ),
+        expected_mean,
+    )
+    assert np.isclose(
+        causalprog.algorithms.standard_deviation(
+            graph, outcome_node_label="X", samples=samples, rng_key=rng_key
+        ),
+        expected_std_dev,
+    )
+
+
+@pytest.mark.parametrize(
+    ("mean", "stdev", "stdev2", "samples", "rtol"),
+    [
+        pytest.param(
+            1.0,
+            1.0,
+            0.8,
+            100,
+            1,
+            id="N(mean=N(mean=0, stdev=1), stdev=0.8), 100 samples",
+        ),
+        pytest.param(
+            3.0,
+            0.5,
+            1.0,
+            10000,
+            1e-1,
+            id="N(mean=N(mean=3, stdev=0.5), stdev=1), 10^4 samples",
+        ),
+        pytest.param(
+            2.0,
+            0.7,
+            0.8,
+            1000000,
+            1e-2,
+            id="N(mean=N(mean=2, stdev=0.7), stdev=0.8), 10^6 samples",
+        ),
+    ],
+)
+def test_mean_stdev_two_node_graph(
+    two_normal_graph, samples, rtol, mean, stdev, stdev2, rng_key
+):
+    if samples > 100000:  # noqa: PLR2004
+        pytest.xfail("Test currently runs out of memory")
```

Comment:
We should work out why setting the number of samples to 10^6 gives an out of memory error in this test

Reply:
I've opened #74 for this